Failure to display peers of a torrent

## Summary

On some torrents peers tab fails to display. I guess it is somehow connected to some specific peer that breaks things. I tested with a torrent and sometimes peers display fine, but sometimes they fail to open.

## Expected Behavior

When you open a torrent and click on peers tab - a list of peers should display.


## Current Behavior

Sometimes (not everytime, because you cannot reproduce it everytime) peers tab fails to display. When it happens if you click on peers tab you cannot see peers tab, but instead you stay on the same tab. 

Example screenshot:

![screenshot_2018-06-15_21-54-48](https://user-images.githubusercontent.com/9135358/41484817-e97bcfd4-70e6-11e8-9299-e700e53f8890.png)

When this happens in Chrome Developer tools you see the below errors:

```
.png$:206 Uncaught Error: Cannot find module './je.png'.
    at a (.png$:206)
    at r (.png$:201)
    at TorrentPeers.js:41
    at Array.map (<anonymous>)
    at t.value (TorrentPeers.js:32)
    at d._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:796)
    at d._renderValidatedComponent (ReactCompositeComponent.js:819)
    at d.performInitialMount (ReactCompositeComponent.js:359)
    at d.mountComponent (ReactCompositeComponent.js:255)
    at Object.mountComponent (ReactReconciler.js:43)
ReactReconciler.js:60 Uncaught (in promise) TypeError: Cannot read property 'getHostNode' of null
    at Object.getHostNode (ReactReconciler.js:60)
    at d.getHostNode (ReactCompositeComponent.js:381)
    at Object.getHostNode (ReactReconciler.js:60)
    at Object.updateChildren (ReactChildReconciler.js:111)
    at m._reconcilerUpdateChildren (ReactMultiChild.js:209)
    at m._updateChildren (ReactMultiChild.js:308)
    at m.updateChildren (ReactMultiChild.js:295)
    at m._updateDOMChildren (ReactDOMComponent.js:944)
    at m.updateComponent (ReactDOMComponent.js:758)
    at m.receiveComponent (ReactDOMComponent.js:720)
ReactReconciler.js:60 Uncaught TypeError: Cannot read property 'getHostNode' of null
    at Object.getHostNode (ReactReconciler.js:60)
    at d.getHostNode (ReactCompositeComponent.js:381)
    at Object.getHostNode (ReactReconciler.js:60)
    at Object.updateChildren (ReactChildReconciler.js:111)
    at m._reconcilerUpdateChildren (ReactMultiChild.js:209)
    at m._updateChildren (ReactMultiChild.js:308)
    at m.updateChildren (ReactMultiChild.js:295)
    at m._updateDOMChildren (ReactDOMComponent.js:944)
    at m.updateComponent (ReactDOMComponent.js:758)
    at m.receiveComponent (ReactDOMComponent.js:720)
```

This errors keep repeating over and over...

## Possible Solution

Do not know.

## Steps to Reproduce (for bugs)
1. Click on a torrent.
2. Start to change detail tabs.
3. Click to peer tab.
4. It does not display.
5. Open chrome developer tools and see errors in console.

## Context

Peers tab is a nice functionality and it should work every time.

## Your Environment

* Version used:
  + v1.0.0
  + commit 1493f8f9cbf6d6c94f1083a53aaba82d95d508af 
* Chromium 67.0.3396.87, NodeJS v10.4.1
* Server (Debian Stretch 9.4 x64), Client (Fully updated Arch Linux x64)

I found one more that is missing:

```.png$:207 Uncaught Error: Cannot find module './cw.png'.
    at a (.png$:207)
    at r (.png$:202)
    at TorrentPeers.js:41
    at Array.map (<anonymous>)
    at t.value (TorrentPeers.js:32)
    at d._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:796)
    at d._renderValidatedComponent (ReactCompositeComponent.js:819)
    at d.performInitialMount (ReactCompositeComponent.js:359)
    at d.mountComponent (ReactCompositeComponent.js:255)
    at Object.mountComponent (ReactReconciler.js:43)
ReactReconciler.js:60 Uncaught (in promise) TypeError: Cannot read property 'getHostNode' of null
    at Object.getHostNode (ReactReconciler.js:60)
    at d.getHostNode (ReactCompositeComponent.js:381)
    at Object.getHostNode (ReactReconciler.js:60)
    at Object.updateChildren (ReactChildReconciler.js:111)
    at m._reconcilerUpdateChildren (ReactMultiChild.js:209)
    at m._updateChildren (ReactMultiChild.js:308)
    at m.updateChildren (ReactMultiChild.js:295)
    at m._updateDOMChildren (ReactDOMComponent.js:944)
    at m.updateComponent (ReactDOMComponent.js:758)
    at m.receiveComponent (ReactDOMComponent.js:720)```

Got another one...

```
Uncaught Error: Cannot find module './mq.png'.
    at a (.png$:208)
    at r (.png$:203)
    at TorrentPeers.js:41
    at Array.map (<anonymous>)
    at t.value (TorrentPeers.js:32)
    at d._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:796)
    at d._renderValidatedComponent (ReactCompositeComponent.js:819)
    at d.performInitialMount (ReactCompositeComponent.js:359)
    at d.mountComponent (ReactCompositeComponent.js:255)
    at Object.mountComponent (ReactReconciler.js:43)
```

--- a/client/src/javascript/components/modals/torrent-details-modal/TorrentPeers.js
+++ b/client/src/javascript/components/modals/torrent-details-modal/TorrentPeers.js
@@ -43,8 +43,7 @@ export default class TorrentPeers extends React.Component {
             let flagImageSrc;
             try {
               flagImageSrc = require(`../../../../images/flags/${countryCode.toLowerCase()}.png`);
-            }
-            catch(e) {
+            } catch {
               this.flagImageAsErrored(countryCode);
               flagImageSrc = null;
             }

--- a/client/src/javascript/components/modals/torrent-details-modal/TorrentPeers.js
+++ b/client/src/javascript/components/modals/torrent-details-modal/TorrentPeers.js
@@ -43,7 +43,7 @@ export default class TorrentPeers extends React.Component {
             let flagImageSrc;
             try {
               flagImageSrc = require(`../../../../images/flags/${countryCode.toLowerCase()}.png`);
-            } catch {
+            } catch (err) {
               this.flagImageAsErrored(countryCode);
               flagImageSrc = null;
             }

--- a/client/src/javascript/components/modals/torrent-details-modal/TorrentPeers.js
+++ b/client/src/javascript/components/modals/torrent-details-modal/TorrentPeers.js
@@ -16,12 +16,14 @@ export default class TorrentPeers extends React.Component {
     };
   }
 
+  flagImageAsErrored(countryCode) {
+    let {erroredCountryImages} = this.state;
+    erroredCountryImages.push(countryCode);
+    this.setState({erroredCountryImages});
+  }
+
   getImageErrorHandlerFn(countryCode) {
-    return () => {
-      let {erroredCountryImages} = this.state;
-      erroredCountryImages.push(countryCode);
-      this.setState({erroredCountryImages});
-    };
+    return () => this.flagImageAsErrored(countryCode);
   }
 
   render() {
@@ -43,7 +45,7 @@ export default class TorrentPeers extends React.Component {
               flagImageSrc = require(`../../../../images/flags/${countryCode.toLowerCase()}.png`);
             }
             catch(e) {
-              this.getImageErrorHandlerFn(countryCode);
+              this.flagImageAsErrored(countryCode);
               flagImageSrc = null;
             }
 

--- a/client/src/javascript/components/modals/torrent-details-modal/TorrentPeers.js
+++ b/client/src/javascript/components/modals/torrent-details-modal/TorrentPeers.js
@@ -38,16 +38,25 @@ export default class TorrentPeers extends React.Component {
           let image = null;
 
           if (!erroredCountryImages.includes(countryCode)) {
-            const flagImageSrc = require(`../../../../images/flags/${countryCode.toLowerCase()}.png`);
+            let flagImageSrc;
+            try {
+              flagImageSrc = require(`../../../../images/flags/${countryCode.toLowerCase()}.png`);
+            }
+            catch(e) {
+              this.getImageErrorHandlerFn(countryCode);
+              flagImageSrc = null;
+            }
 
-            image = (
-              <img
-                alt={countryCode}
-                className="peers-list__flag__image"
-                onError={this.getImageErrorHandlerFn(countryCode)}
-                src={flagImageSrc}
-              />
-            );
+            if (flagImageSrc) {
+              image = (
+                <img
+                  alt={countryCode}
+                  className="peers-list__flag__image"
+                  onError={this.getImageErrorHandlerFn(countryCode)}
+                  src={flagImageSrc}
+                />
+              );
+            }
           }
 
           peerCountry = (

